pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -3199,3 +3207,53 @@ func handleDeviceOperation(ctxPtr *zedagentContext, op types.DeviceOperation) {
 	shutdownAppsGlobal(ctxPtr)
 	// nothing else to be done
 }
+
+func parseEdgeNodeClusterConfig(getconfigCtx *getconfigContext,
+	config *zconfig.EdgeDevConfig) {
+
+	ctx := getconfigCtx.zedagentCtx
+	zcfgCluster := config.GetCluster()
+	if zcfgCluster == nil {
+		log.Functionf("parseEdgeNodeClusterConfig: No EdgeNodeClusterConfig, Unpublishing")
+		pub := ctx.pubEdgeNodeClusterConfig
+		items := pub.GetAll()
```

Comment:
Is there only a global key for this? If so there is no need to call GetAll - just call Unpublish("global")

Otherwise call GetAll and walk the set of returned objects and unpublish each of their keys.

Reply:
Ok.